src/NotificationEventMailing.php

```diff
@@ -148,6 +148,24 @@ public static function send(array $data)
                     )
                 );
 
+                // Add custom header (References) for mail grouping in reader (for Gmail)
+                // see https://workspaceupdates.googleblog.com/2019/03/threading-changes-in-gmail-conversation-view.html
+                $mmail->AddCustomHeader(
+                    str_replace(
+                        [
+                            '%uuid',
+                            '%itemtype',
+                            '%items_id'
+                        ],
+                        [
+                            Config::getUuid('notification'),
+                            $current->fields['itemtype'],
+                            $current->fields['items_id']
+                        ],
```

Comment:
You could mutualize code with `In-Reply-To:`:

```php
$email_ref = sprintf('<GLPI-%s-%s-%s>', Config::getUuid('notification'), $current->fields['itemtype'], $current->fields['items_id']);
$mmail->AddCustomHeader("In-Reply-To: $email_ref");
$mmail->AddCustomHeader("References: $email_ref");

Reply:
done